core/dimensions.cpp

```diff
@@ -199,22 +199,22 @@ int32_t Dimensions::index(const Dim dim) const {
 /// Return the direct sum, i.e., the combination of dimensions in a and b.
 ///
 /// Throws if there is a mismatching dimension extent.
-Dimensions merge(const Dimensions &a, const Dimensions &b) {
+Dimensions merge(const Dimensions &a, const Dimensions &b, bool ignoreExtents) {
```

Comment:
Is this bool argument really necessary? I feel that it is confusing and should generally be avoided.

Reply:
Agreed. I would not have left it in this state.